### PR TITLE
Write readNetworkingV1NamespacedIngressStatus test - +1 endpoint coverage

### LINKED ISSUE #93035: Write readNetworkingV1NamespacedIngressStatus test+promote - +1 endpoint coverage
# Progress <code>[1/6]</code>

-   [X] Test in Ginkgo : [ingress.go](https://github.com/ii/kubernetes/blob/fix-ingress-status/test/e2e/network/ingress.go#L1145-L1151)
-   [X] test approval issue : [kubernetes/kubernetes#93035](https://github.com/kubernetes/kubernetes/issues/93035)
-   [X] test pr : [kuberenetes/kubernetes#93038](https://github.com/kubernetes/kubernetes/pull/93038)
-   [ ] two weeks soak start date : testgrid-link
-   [ ] two weeks soak end date :
-   [ ] test promotion pr : kubernetes/kubernetes#?

# Identifying an untested feature Using APISnoop.
It was picked up in APISnoop that an untested endpoint was promoted from Beta to GA.
PR [#91996](https://github.com/kubernetes/kubernetes/pull/91996) ingress: promote CRUD API tests for v1 to conformance covered the Ingress life cycle tests, but **readNetworkingV1NamespacedIngressStatus**  was missed.

According to this APIsnoop **readNetworkingV1NamespacedIngressStatus** endpoint is untested.

# The test updated in Ingress.go

## Test the functionality in Ginko

		ginkgo.By("get /status")
		ingResource := schema.GroupVersionResource{Group: "networking.k8s.io", Version: ingVersion, Resource: "ingresses"}
		gottenStatus, err := f.DynamicClient.Resource(ingResource).Namespace(ns).Get(context.TODO(), createdIngress.Name, metav1.GetOptions{}, "status")
		framework.ExpectNoError(err)
		statusUID, _, err := unstructured.NestedFieldCopy(gottenStatus.Object, "metadata", "uid")
		framework.ExpectNoError(err)
		framework.ExpectEqual(string(createdIngress.UID), statusUID, fmt.Sprintf("createdIngress.UID: %v expected to match statusUID: %v ", createdIngress.UID, statusUID))
'''
# Verifying increase in coverage

List endpoints hit by the test:
readNetworkingV1NamespacedIngressStatus endpoint

# Final notes

If this test get merged, ****test coverage will go up by 1 point****

This test is also created with the goal of conformance promotion.

---

/sig testing
/sig architecture
/area conformance

### PR DESCRIPTION
**Pull request by riaankleinhans**
**What** type of PR is this?
/kind cleanup

**What** this PR does / why we need it:
This PR adds a test to test the following untested endpoints:
**readNetworkingV1NamespacedIngressStatus**

**Which** issue(s) this PR fixes:
Fixes #93035 

**Special** notes for your reviewer:
Adds +1 endpoint test coverage (good for conformance)

**Does** this PR introduce a user-facing change?:
```
NONE

```

**Release note:**
```release-note
NONE
```

**Additional** documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```
NONE
```

/sig testing
/sig architecture
/area conformance
/area testing
/sig network
/sig api-machinery
/sig apps

### CODE REVIEW

File: test/e2e/network/ingress.go [hunk 1]
```diff
@@ -1141,6 +1142,14 @@ var _ = SIGDescribe("Ingress API", func() {
 		framework.ExpectNoError(err)
 		framework.ExpectEqual(updatedStatus.Status.LoadBalancer, statusToUpdate.Status.LoadBalancer, fmt.Sprintf("updated object expected to have updated loadbalancer status %#v, got %#v", statusToUpdate.Status.LoadBalancer, updatedStatus.Status.LoadBalancer))
 
+		ginkgo.By("get /status")
```
Comment (cmluciano) [inline test/e2e/network/ingress.go:1145]: Nit: Comment here saying what this test is for

File: test/e2e/network/ingress.go [hunk 2]
```diff
@@ -1141,6 +1142,14 @@ var _ = SIGDescribe("Ingress API", func() {
...
+		ingResource := schema.GroupVersionResource{Group: "networking.k8s.io", Version: ingVersion, Resource: "ingresses"}
+		gottenStatus, err := f.DynamicClient.Resource(ingResource).Namespace(ns).Get(context.TODO(), createdIngress.Name, metav1.GetOptions{}, "status")
```
Comment (cmluciano) [inline test/e2e/network/ingress.go:1147]: Did you find that the DynamicClient was the only way to use the Namespace call?

File: test/e2e/network/ingress.go [hunk 3]
```diff
@@ -1141,6 +1142,14 @@ var _ = SIGDescribe("Ingress API", func() {
...
+		framework.ExpectNoError(err)
+		statusUID, _, err := unstructured.NestedFieldCopy(gottenStatus.Object, "metadata", "uid")
+		framework.ExpectNoError(err)
+		framework.ExpectEqual(string(createdIngress.UID), statusUID, fmt.Sprintf("createdIngress.UID: %v expected to match statusUID: %v ", createdIngress.UID, statusUID))
```
Comment (cmluciano) [inline test/e2e/network/ingress.go:1151]: Is the string formatting needed here over something like [this](https://github.com/kubernetes/kubernetes/blob/31962804c0e0fc12319437febd46c9ce45f4ef0e/test/e2e/network/ingress.go#L1066)

**Review by cmluciano — COMMENTED** [review]
Sorry for the delayed review. I was having a few local dev environment issues.

File: test/e2e/network/ingress.go (hunk 3 above)
Comment (heyste) [reply test/e2e/network/ingress.go:1151]: I tried the other option first but found that `createdIngress.UID` was `types.UID` when trying to match it with `statusUID` which was `string`

File: test/e2e/network/ingress.go (hunk 2 above)
Comment (heyste) [reply test/e2e/network/ingress.go:1147]: Using the DynamicClient this way the first option that worked for me. Would you prefer it done another way?

File: test/e2e/network/ingress.go (hunk 1 above)
Comment (heyste) [reply test/e2e/network/ingress.go:1145]: Was following the same pattern that's already in the test 
- Line 1123 `ginkgo.By("patching /status")`
- Line 1136 `ginkgo.By("updating /status")`

**Comment by Riaankl** [comment]
@aojea can you please review, we believe it is ready to merge

File: test/e2e/network/ingress.go (hunk 2 above)
Comment (aojea) [reply test/e2e/network/ingress.go:1147]: is the Get not working for this?
https://github.com/kubernetes/kubernetes/blob/30b0ebd6d43de3b9eb6b78d877612a4d7aed85a8/test/e2e/network/ingress.go#L1063

Comment (liggitt) [reply test/e2e/network/ingress.go:1147]: that does not get the status subresource

File: test/e2e/network/ingress.go (hunk 1 above)
Comment (liggitt) [reply test/e2e/network/ingress.go:1145]: this looks fine to me as is

**Comment by liggitt** [comment]
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/93038/pull-kubernetes-node-e2e/1283208470891335680 flake was #93210

**Pull request merged**